File: 219.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> mm;
        for (int i=0; i<nums.size(); i++){
           if(mm.find(nums[i]) != mm.end()){
               if(i-mm[nums[i]]<=k){
                   return true;
               }
           }
           mm[nums[i]] = i;
        }
        return false;
    }
};

File: go-solutions/219.go
import "math"

func containsNearbyDuplicate(nums []int, k int) bool {
	mm := make(map[int]int)
	for i, v := range nums {
		_, ok := mm[v]
		if ok == true {
			t := math.Abs(float64(mm[v] - i))
			if t <= float64(k) {
				return true
			}
		}
		mm[v] = i
	}
	return false
}